sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetRowConverter.scala

```diff
@@ -296,6 +317,15 @@ private[parquet] class ParquetRowConverter(
           }
         }
 
+      case DateType if rebaseDateTime =>
+        new ParquetPrimitiveConverter(updater) {
+          override def addInt(value: Int): Unit = {
+            val rebased = DateTimeUtils.rebaseJulianToGregorianDays(value)
+            // DateType is not specialized in `SpecificMutableRow`, have to box it here.
+            updater.set(rebased.asInstanceOf[DateType#InternalType])
```

Comment:
what's the error if we simply write `updater.set(rebased)` here?

Reply:
No error, I will remove `.asInstanceOf[DateType#InternalType]`